packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -49,250 +49,306 @@ const password = 'password123';
 const commonConfig = { chain: 'goerli', hardfork: 'berlin' };
 
 describe('KeyringController', () => {
-  let keyringController: KeyringController;
-  let preferences: {
-    setAccountLabel: sinon.SinonStub;
-    removeIdentity: sinon.SinonStub;
-    syncIdentities: sinon.SinonStub;
-    updateIdentities: sinon.SinonStub;
-    setSelectedAddress: sinon.SinonStub;
-  };
-  let initialState: {
-    isUnlocked: boolean;
-    keyringTypes: string[];
-    keyrings: Keyring[];
-  };
-  const encryptor = new MockEncryptor();
-  const additionalKeyrings = [QRKeyring];
-  const additionalKeyringBuilders = additionalKeyrings.map((keyringType) =>
-    keyringBuilderFactory(keyringType),
-  );
-  const baseConfig: Partial<KeyringConfig> = {
-    encryptor,
-    keyringBuilders: additionalKeyringBuilders,
-  };
-
-  beforeEach(async () => {
-    preferences = {
-      setAccountLabel: sinon.stub(),
-      removeIdentity: sinon.stub(),
-      syncIdentities: sinon.stub(),
-      updateIdentities: sinon.stub(),
-      setSelectedAddress: sinon.stub(),
-    };
-    keyringController = new KeyringController(preferences, baseConfig);
-
-    initialState = await keyringController.createNewVaultAndKeychain(password);
-  });
-
   afterEach(() => {
     sinon.restore();
   });
 
-  it('should set default state', () => {
-    expect(keyringController.state.keyrings).not.toStrictEqual([]);
-    const keyring = keyringController.state.keyrings[0];
-    expect(keyring.accounts).not.toStrictEqual([]);
-    expect(keyring.index).toStrictEqual(0);
-    expect(keyring.type).toStrictEqual('HD Key Tree');
+  it('should set default state', async () => {
+    await withController({}, async ({ controller }) => {
+      expect(controller.state.keyrings).not.toStrictEqual([]);
+      const keyring = controller.state.keyrings[0];
+      expect(keyring.accounts).not.toStrictEqual([]);
+      expect(keyring.index).toStrictEqual(0);
+      expect(keyring.type).toStrictEqual('HD Key Tree');
+    });
   });
 
   describe('addNewAccount', () => {
     describe('when accountCount is not provided', () => {
       it('should add new account', async () => {
-        const { keyringState: currentKeyringMemState, addedAccountAddress } =
-          await keyringController.addNewAccount();
-        expect(initialState.keyrings).toHaveLength(1);
-        expect(initialState.keyrings[0].accounts).not.toStrictEqual(
-          currentKeyringMemState.keyrings[0].accounts,
-        );
-        expect(currentKeyringMemState.keyrings[0].accounts).toHaveLength(2);
-        expect(initialState.keyrings[0].accounts).not.toContain(
-          addedAccountAddress,
-        );
-        expect(addedAccountAddress).toBe(
-          currentKeyringMemState.keyrings[0].accounts[1],
+        await withController(
+          {},
```

Comment:
Nit: For the `withController` function you copied (good call by the way!), the first argument is optional, so you could leave this out to make this simpler. Same goes for the other tests.